app/scripts/components/common/blocks/block-map.tsx

```diff
@@ -197,57 +193,39 @@ function MapBlock(props: MapBlockProps) {
 
   const [, setProjection] = useState(projectionStart);
 
-  const dataset = datasetId ? datasets[datasetId] : null;
-
-  const resolverBag = useMemo<DatasetDatumFnResolverBag>(
-    () => ({
-      datetime: selectedDatetime,
-      compareDatetime: selectedCompareDatetime,
-      dateFns
-    }),
-    [selectedDatetime, selectedCompareDatetime]
-  );
-
-  const [baseLayerResolvedData] = useMemo(() => {
-    if (!baseLayers || baseLayers.length !== 1) return [null, null];
+  const baseDataLayer:(VizDatasetSuccess| null) = useMemo(() => {
+    if (!baseLayers || baseLayers.length !== 1) return null;
     const baseLayer = baseLayers[0];
 
-    if (baseLayer.status !== S_SUCCEEDED) return [null, null];
+    if (baseLayer.status !== S_SUCCEEDED) return null;
+    return {
+      ...baseLayer,
+      settings: {
+        isVisible: true,
+        opacity: 100
+      }
+    };
+  }, [baseLayers]);
 
-    const bag = { ...resolverBag, raw: baseLayer.data };
-    const data = resolveConfigFunctions(baseLayer.data, bag);
-    return [data];
-  }, [baseLayers, resolverBag]);
-
-  const baseDataLayer: TimelineDataset | null = {
-    data: baseLayerResolvedData,
-    settings: {
-      isVisible: true,
-      opacity: 100
-    }
-  } as unknown as TimelineDataset;
-  const baseTimeDensity = baseLayerResolvedData?.timeDensity;
+  const baseTimeDensity = baseDataLayer?.data.timeDensity;
 
   // Resolve data needed for the compare layer once it is loaded.
-  const [compareLayerResolvedData] = useMemo(() => {
-    if (!compareLayers || compareLayers.length !== 1) return [null, null];
+  const compareDataLayer: (VizDatasetSuccess | null) = useMemo(() => {
+    if (!compareLayers || compareLayers.length !== 1) return null;
     const compareLayer = compareLayers[0];
 
-    if (compareLayer.status !== S_SUCCEEDED) return [null, null];
-    // Include access to raw data.
-    const bag = { ...resolverBag, raw: compareLayer.data };
-    const data = resolveConfigFunctions(compareLayer.data, bag);
-    return [data];
-  }, [compareLayers, resolverBag]);
-
-  const compareDataLayer: TimelineDataset | null = {
-    data: compareLayerResolvedData,
-    settings: {
-      isVisible: true,
-      opacity: 100
-    }
-  } as unknown as TimelineDataset;
-  const compareTimeDensity = compareLayerResolvedData?.timeDensity;
+    if (compareLayer.status !== S_SUCCEEDED) return null;
+
+    return {
+      ...compareLayer,
+      settings: {
+        isVisible: true,
+        opacity: 100
+      }
+    };
+  }, [compareLayers]);
```

Comment:
⛏️ : Just a though to make it less redundant. Combine baseDataLayer and compareDataLayer logic into a function instead?

something like...
```  
const getDataLayer = (layers: VizDataset[] | undefined): (VizDatasetSuccess | null) => {
    if(!layers || layers.length !== 1) return null;
    const someLayer = layers[0];

    if(someLayer.status !== S_SUCCEEDED) return null;
    return {
      ...someLayer,
      settings: {
        isVisible: true,
        opacity: 100
      },
    }
  };

  const baseDataLayer = useMemo(() => getDataLayer(baseLayers), [baseLayers])
  const compareDataLayer = useMemo(() => getDataLayer(compareLayers), [compareLayers])
  ```
  
  But current way is also fine which is just more explicit. whichever

Reply:
This is a great suggestion. I also made some changes so that both layers can be fetched with one call of `  useReconcileWithStacMetadata(layersToFetch, setLayers);`, and combined with your suggestion.